modules/core/shared/src/main/scala/org/legogroup/woof/EnclosingClass.scala

```diff
@@ -0,0 +1,20 @@
+package org.legogroup.woof
+
+import scala.annotation.tailrec
+
+case class EnclosingClass(fullName: String, lineLength: Int = 80):
+  def printableName: String = reduceNameLength(fullName)
+
+  @tailrec
+  private def reduceNameLength(name: String, chomps: Int = 0): String = {
+    val names = name.split('.')
+    val isTooLong = name.length > lineLength
+    val notLastPossibleChomp = chomps < names.length - 1
+    val keepChomping = notLastPossibleChomp && isTooLong
+    if (keepChomping) {
+      names.update(chomps, names(chomps).head.toString)
+      reduceNameLength(names.mkString("."), chomps + 1)
+    } else {
+      name
+    }
```

Comment:
```suggestion
  private def reduceNameLength(name: String): String =
    List
      .unfold(name.split('.').toList) {
        case Nil                                                   => none
        case last :: Nil                                           => (last, Nil).some
        case parts @ x :: xs if parts.mkString.length > lineLength => (x.take(1), xs).some
        case x :: xs                                               => (x, xs).some
      }
      .mkString(".")
```

Reply:
TIL `unfold`. 

The only thing that jumps out to me is that we're building the string back up without a delimiter before checking if it is too long.  The original representation is actually `parts.mkString.length + length - 1` long if the delimiters are included.  Do you think we should just remake the string with the separator to keep what we're cutting down consistent and not dependent on package depth? Ill take your commit, and I can add an argument to `mkString` if you agree.